docs/architecture.md

```diff
@@ -0,0 +1,83 @@
+# Architecture
+
+⚠️  The following is about the _intended_ architecture of gov.uk/alerts, which may not (yet) reflect reality.
+
+The overall architecture looks like this:
+
+```
+                                     +-------------------------------+
+                                     |           data.yaml           |
+                                     |                               |
+                                     |    (legacy static alerts)     |
+                                     +-------------------------------+
+                                       |
+                                       |
+                                       v
++------------------------------+     +-------------------------------+     +--------------------------+
+|                              |     |                               |     |        Notify API        |
+| static pages (e.g. homepage) |     |     common rendering code     |     |                          |
+|                              | --> |                               | <-- | dynamic alerts (from DB) |
++------------------------------+     +-------------------------------+     +--------------------------+
+                                       |                                     |
+                                       |                                     |
+                                       v                                     |
+                                     +-------------------------------+       |
+                                     |     Celery task (on PaaS)     |       |
+                                     |                               |       |
+                                     | (also triggered by Concourse) | <-----+
+                                     +-------------------------------+
+                                       |
+                                       |
+                                       v
++------------------------------+     +-------------------------------+
+|   static assets (JS, CSS)    |     |                               |
+|                              |     |           S3 bucket           |
+|   (uploaded by Concourse)    | --> |                               |
++------------------------------+     +-------------------------------+
+                                       |
+                                       |
+                                       v
+                                     +-------------------------------+
+                                     |          CloudFront           |
+                                     |                               |
+                                     |   (tweaks caching headers)    |
```

Comment:
Should we note that this is performed by Lambda@Edge response function?

Reply:
I'd prefer not to here, as this is mostly to provide context about this repo. Mentioning that it "tweaks caching headers" should be enough of a prompt for someone to go looking the Terraform if they're interested.

We could always add a README to the Terraform module so it explains itself in more detail. Incidentally, I just found out about `terraform graph`, but I'm not sure anyone would ever want to see this 😬 :

![test](https://user-images.githubusercontent.com/9029009/129934894-eb5ee32d-abea-4b10-b9e8-4b9743f9cfc4.png)